From 58a2d817a82100d287c60c63315d81445cdba3f9 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Thu, 15 Jun 2017 15:12:08 -0400
Subject: [PATCH] cgroups: work around issue in gcc 7

This works around
https://bugzilla.yoctoproject.org/show_bug.cgi?id=11672
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78969

By removing a single cgroup entry. For the majority of usecases this
loss of a single entry should not be an issue and once gcc 7 is fixed
we can revert this.

Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
---
 src/lxc/cgroups/cgfsng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index ebd548b..c520abd 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -1373,7 +1373,7 @@ static inline bool cgfsng_create(void *hdata)
 	offset = cgname + len - 5;
 
 again:
-	if (idx == 1000) {
+	if (idx == 999) {
 		ERROR("Too many conflicting cgroup names");
 		goto out_free;
 	}
-- 
2.7.4

